src/coreclr/src/pal/src/init/pal.cpp

```diff
@@ -1289,116 +1280,6 @@ bool GetAbsolutePath(const char* path, PathCharString& absolutePath)
     return result;
 }
 
-bool GetEntrypointExecutableAbsolutePath(PathCharString& entrypointExecutable)
-{
-    bool result = false;
-
-    entrypointExecutable.Clear();
-
-    // Get path to the executable for the current process using
-    // platform specific means.
-#if defined(__APPLE__)
-
-    // On Mac, we ask the OS for the absolute path to the entrypoint executable
-    uint32_t lenActualPath = 0;
-    if (_NSGetExecutablePath(nullptr, &lenActualPath) == -1)
-    {
-        // OSX has placed the actual path length in lenActualPath,
-        // so re-attempt the operation
-        PathCharString resizedPath;
-        char *pResizedPath = resizedPath.OpenStringBuffer(lenActualPath);
-        if (_NSGetExecutablePath(pResizedPath, &lenActualPath) == 0)
-        {
-            resizedPath.CloseBuffer(lenActualPath - 1);
-            entrypointExecutable.Set(resizedPath);
-            result = true;
-        }
-    }
-#elif defined (__FreeBSD__)
-    static const int name[] =
-    {
-        CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1
-    };
-    char path[PATH_MAX];
-    size_t len;
-
-    len = sizeof(path);
-    if (sysctl(name, 4, path, &len, nullptr, 0) == 0)
-    {
-        entrypointExecutable.Set(path, len);
-        result = true;
-    }
-    else
-    {
-        // ENOMEM
-        result = false;
-    }
-#elif defined(__NetBSD__) && defined(KERN_PROC_PATHNAME)
-    static const int name[] =
-    {
-        CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME,
-    };
-    char path[MAXPATHLEN];
-    size_t len;
-
-    len = sizeof(path);
-    if (sysctl(name, __arraycount(name), path, &len, NULL, 0) != -1)
-    {
-        entrypointExecutable.Set(path, len);
-        result = true;
-    }
-    else
-    {
-        result = false;
-    }
-#elif defined(__sun)
-    const char *path;
-    if ((path = getexecname()) == NULL)
-    {
-        result = false;
-    }
-    else if (*path != '/')
-    {
-        char *cwd;
-        if ((cwd = getcwd(NULL, PATH_MAX)) == NULL)
-        {
-            result = false;
-        }
-        else
-        {
-            entrypointExecutable.Set(cwd, strlen(cwd));
-            entrypointExecutable.Append('/');
-            entrypointExecutable.Append(path, strlen(path));
-
-            result = true;
-            free(cwd);
-        }
-    }
-    else
-    {
-        entrypointExecutable.Set(path, strlen(path));
-        result = true;
-    }
-#else
-
-#if HAVE_GETAUXVAL && defined(AT_EXECFN)
-    const char *execfn = (const char *)getauxval(AT_EXECFN);
```

Comment:
This was added to in https://github.com/dotnet/runtime/pull/40791 , but it is not in the shared version. Is the problem fixed in #40791 going to be back?

Reply:
I simply used the libraries version, because it was the latest.

#40791  is not particularly fixing anything with getauxval, that PR has added the functionlity from unixruncommon to resolve the current exe paths in general. The original PR that added getauxval was https://github.com/dotnet/coreclr/pull/24696; it was done for optimization (to avoid a system call). This works well without that optimization. I can add it back if it is necessary.

Reply:
Can you please add it to the common code. The coreclr stuff was the most up to date, when the getauxval version was added, there were still corefx / installer separate repos and it was missed in the other two. The benefit of this way of getting the current exe path is that it doesn't need the /proc file system, so it works in chroot scenarios and also in some other important scenarios.